app/workers/travis_branch_monitor.rb

```diff
@@ -0,0 +1,104 @@
+require 'travis'
+
+class TravisBranchMonitor
+  include Sidekiq::Worker
+  sidekiq_options :queue => :miq_bot_glacial, :retry => false
+
+  include Sidetiq::Schedulable
+  recurrence { hourly.minute_of_hour(0, 15, 30, 45) }
+
+  include SidekiqWorkerMixin
+
+  class << self
+    private
+
+    # For this class, repos will sometimes the repo needs to be mapped to a
+    # specific gitter room, so a hash is required.
+    #
+    # This override allows for doing this in the config
+    #
+    #   travis_branch_monitor:
+    #     included_repos:
+    #       ManageIQ/manageiq-ui-classic: ManageIQ/ui
+    #       ManageIQ/manageiq-gems-pending: ManageIQ/core
+    #       ManageIQ/manageiq:
+    #       ManageIQ/miq_bot:
+    #
+    # Which you are allowed to leave the value empty, and the key will be used
+    # where appropriate (not used in this class).
+    #
+    # The result from the above for this method will then be:
+    #
+    #   [
+    #     "ManageIQ/manageiq-ui-classic",
+    #     "ManageIQ/manageiq-gems-pending",
+    #     "ManageIQ/manageiq",
+    #     "ManageIQ/miq_bot"
+    #   ]
+    #
+    def included_and_excluded_repos
+      super # just used for error handling...
+
+      [
+        settings.included_repos.try(:to_h).try(:stringify_keys).try(:keys),
+        settings.excluded_repos.try(:to_h).try(:stringify_keys).try(:keys)
+      ]
+    end
+  end
+
+  def perform
+    if !first_unique_worker?
+      logger.info("#{self.class} is already running, skipping")
+    else
+      process_repos
+    end
+  end
+
+  def process_repos
+    enabled_repos.each do |repo|
+      process_repo(repo)
+    end
+  end
+
+  def process_repo(repo)
+    # repo.regular_branch_names.each do |branch|
+    # end
+
+    # TODO:  Handle this for all release branches
+    process_branch(repo, 'master')
+  end
+
+  def process_branch(repo, branch)
+    branch_record = repo.regular_branches.where(:name => branch).first
+
+    # If we already have a failure record, call notify with that record
+    return branch_record.notify_of_failure if branch_record.previously_failing?
+
+    # otherwise, check if any builds exist with a failures, and if so, create a
+    # new BuildFailure record.
```

Comment:
Some Basmati Rice in the comments here and above now that you don't have a separate table.

Reply:
Thanks.  👍 

Tried to catch some of this when I was refactoring yesterday, but clearly missed some spots.